src/components/popover/popover.tsx

```diff
@@ -114,7 +114,7 @@ export class Popover {
     private setupClickHandler() {
         if (this.open) {
             document.addEventListener('click', this.globalClickListener, {
-                passive: true,
+                capture: true,
```

Comment:
`capture: true` makes this listener activate in the "capture" phase of the event lifecycle. Quirksmode has a [good article](https://www.quirksmode.org/js/events_order.html) on the difference between event capturing and event bubbling.

In short, it means our global click lister will be called before any click handlers that are attached to any other elements. Because of that, we get the chance to detect if the click was outside the popover, close the popover and stop the event from propagating further, before the event even reaches any other components. This acts like a scrim, without requiring any extra element to catch the click.

Reply:
The reason I removed `passive: true` is that I don't want to risk the event propagating before we get the chance to stop it. It did work even with `passive: true`, but it felt a bit paradoxical, so I removed it to not risk any race conditions. According to the [documentation](https://stenciljs.com/docs/events#passive), Stencil will make the listener passive if it can detect that it can do so, so this listener may be made passive by Stencil anyway. And, passive listeners are mostly important when listening to scroll-events anyway. A used won't _click_ hundreds of times a second 😅

Reply:
The passive stuff from the Stencil docs probably only applies if the `@Listen` decorator is used, not if we use `addEventListener` directly :slightly_smiling_face:

Reply:
![facepalm-sequence](https://user-images.githubusercontent.com/154725/108379356-1f928b80-7206-11eb-8b51-4d6d813b369e.gif)

Yeah, obviously 😂